payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -63,19 +65,14 @@ internal class GooglePayLauncherActivity : AppCompatActivity() {
             }
         }
 
-        if (!viewModel.hasLaunched) {
-            lifecycleScope.launch {
-                viewModel.createLoadPaymentDataTask().fold(
```

Comment:
This was the main problem. We call a suspending function from the activity.

Now this suspending fun is called in the view model when it's initialized, and the activity listens for the result when it's resumed, and launched the Google Pay task when it's ready.